### PR TITLE
Multiple choice page

### PR DESCRIPTION
**Pull request by dbasak763**
Title: Multiple‑choice question phase + transition to typing

User Story
As a player, I can answer a timed multiple‑choice question and then move into the typing phase, so gameplay flows between phases.

Summary (real project code)

Builds the MCQ phase UI with prompt + answer options.
Adds timer + selection feedback for interaction clarity.
Routes/advances into typing phase (auto/continue).
Integrates with question data flow (e.g., question sets).
Approach / Architecture

Dedicated question phase component with clear state handling.
Transition logic separated from display logic for readability.
Data loading isolated from UI rendering.
Tests (deterministic + mocked where needed)

Success paths (valid selection + transition)
Failure/edge cases (timeout, invalid selection, missing data)
External services mocked as needed (e.g., Firestore)
How to run: npm test

### CODE REVIEW

**Review by Aifrion — APPROVED** [review]
Great job looks well done to me! There may be a conflict when we try to merge the authentication branch though, but that should not be too hard to resolve

File: client/src/pages/Question/index.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    options,
+    correctIndex: raw.correctAnswer ?? 0,
+  };
+}
+
+export default function Question() {
+  const { code } = useParams<{ code: string }>();
+  const navigate = useNavigate();
+  const [questionSets, setQuestionSets] = useState<any[]>([]);
+  const [currentIndex, setCurrentIndex] = useState(0);
```
Comment (AshM-4) [inline client/src/pages/Question/index.tsx]: We aren't incrementing the currentIndex after initialization. Both the answer and timeout handlers only handle the typing navigation. So any question set or even the current one with multiple questions breaks as it will only show the first question.

File: client/src/pages/Question/index.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  }, []);
+
+  const advanceToTyping = useCallback(() => {
+    if (hasNavigatedRef.current) return;
+    hasNavigatedRef.current = true;
+    if (autoAdvanceTimeoutRef.current) {
+      clearTimeout(autoAdvanceTimeoutRef.current);
+    }
+    const typingCode = code ?? "practice";
+    navigate(`/typing/${typingCode}`);
```
[the remark this answers is not in the document]
Comment (AshM-4) [reply client/src/pages/Question/index.tsx]: Agreed. Also instead of passing the question from the Question page, if we want to prioritize decoupling we should make an independent call to the database from the Game component to get the correct typing sentence.

File: client/src/pages/Question/components/AnswerCard.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  default:
+    'border-slate-200 bg-white shadow hover:-translate-y-[1px] hover:shadow-md focus:ring-purple-300 text-slate-800',
+  correct: 'border-green-500 bg-green-500 shadow text-white',
+  wrong: 'border-red-500 bg-red-500 shadow text-white',
+};
+
+export default function AnswerCard({
+  label,
+  text,
+  color: _color,
```
[the remark this answers is not in the document]
Comment (beihaogu) [reply client/src/pages/Question/components/AnswerCard.tsx]: No.
